### PR TITLE
Add Geant4

### PR DESCRIPTION
**Pull request by jstrube**
Geant4 is a library that simulates the transport of elementary particles through matter. It is frequently used in high energy physics, medical physics and space science.
This builder unfortunately times out on travis, but it works for me locally.

### CODE REVIEW

File: G/Geant4/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+make install
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = [
+    Linux(:x86_64, libc=:glibc),
+    MacOS(:x86_64),
+    FreeBSD(:x86_64)
+]
```
Comment (staticfloat) [inline G/Geant4/build_tarballs.jl]: I take it this doesn't build on Windows?  What about i686 Linux, or musl?

Comment (giordano) [reply G/Geant4/build_tarballs.jl]: I think it should work with 32-bit platforms, on the website the pre-built library is actually for 32-bit windows: https://geant4.web.cern.ch/support/download

**Comment by jstrube** [comment]
Didn't work for me... That doesn't mean it can't be made to work in the
future. The GEANT4 project page has binaries for Windows, so it must be
possible. I don't have the bandwidth to figure this out.

If @oschulz gets his bindings working, maybe some people start using this
on Windows.

**Comment by oschulz** [comment]
> If @oschulz gets his bindings working

I will take a look at that again, but it may be a while - first have to bring ROOT.jl back, and get some other things finished ... so if someone else wants to jump in here in the mean time, please feel free: https://github.com/JuliaHEP/Geant4.jl/tree/dev

**Pull request merged**